pymc3/tests/test_distributions.py

```diff
@@ -1724,6 +1740,174 @@ def test_batch_multinomial(self):
         sample = dist.random(size=2)
         assert_allclose(sample, np.stack([vals, vals], axis=0))
 
+    @pytest.mark.parametrize("n", [2, 3])
+    def test_dirichlet_multinomial(self, n):
+        self.pymc3_matches_scipy(
+            DirichletMultinomial,
+            Vector(Nat, n),
+            {"a": Vector(Rplus, n), "n": Nat},
+            dirichlet_multinomial_logpmf,
+        )
+
+    def test_dirichlet_multinomial_matches_beta_binomial(self):
+        a, b, n = 2, 1, 5
+        ns = np.arange(n + 1)
+        ns_dm = np.vstack((ns, n - ns)).T  # covert ns=1 to ns_dm=[1, 4], for all ns...
+        bb_logp = pm.BetaBinomial.dist(n=n, alpha=a, beta=b).logp(ns).tag.test_value
+        dm_logp = pm.DirichletMultinomial.dist(n=n, a=[a, b]).logp(ns_dm).tag.test_value
+        dm_logp = dm_logp.ravel()
+        assert_allclose(bb_logp, dm_logp)
+
+    @pytest.mark.parametrize(
+        "a, n",
+        [
+            [[0.25, 0.25, 0.25, 0.25], 1],
+            [[0.3, 0.6, 0.05, 0.05], 2],
+            [[0.3, 0.6, 0.05, 0.05], 10],
+        ],
+    )
+    def test_dirichlet_multinomial_mode(self, a, n):
+        _a = np.array(a)
+        with Model() as model:
+            m = DirichletMultinomial("m", n, _a, _a.shape)
+        assert_allclose(m.distribution.mode.eval().sum(), n)
+        _a = np.array([a, a])
+        with Model() as model:
+            m = DirichletMultinomial("m", n, _a, _a.shape)
+        assert_allclose(m.distribution.mode.eval().sum(axis=-1), n)
+
+    @pytest.mark.parametrize(
+        "a, shape, n",
+        [
+            [[0.25, 0.25, 0.25, 0.25], 4, 2],
+            [[0.25, 0.25, 0.25, 0.25], (1, 4), 3],
+            # 3: expect to fail
```

Comment:
Why not test that it is indeed failing?

Reply:
> Why not test that it is indeed failing?

Are you talking about the third and fifth parameterizations to `test_dirichlet_multinomial_random`? This was copied directly from Multinomial so I'm also not actually sure why those two tests are expected to fail (in fact I'm not intuiting why they should fail at first glance)...

I can (1) try running them and see what happens, and (2) check the git blame and pull in the whomever added that comment in the first place.

Reply:
Maybe let's start with (1). The reason I ask is that if it fails we can test that directly. If it doesn't maybe we should do (2)

Reply:
One of them passed as written. The other was missing the `n` parameter, which I added as `[2]*10` to test what I think is a slightly different combination of parameter shapes, which now passes as well.

Reply:
Does it also pass on the Multinomial?

Reply:
Yes, both pass as re-written for the DM (i.e. adding the `n` parameter I mentioned above).